mdp/nodes/recursive_expansion_nodes.py

```diff
@@ -0,0 +1,623 @@
+        :rtype: int
+        """
+        res = super(RecursiveExpansionNode, self).expanded_dim(num_vars)
+        return res+1 if self.with0 else res
+
+    def _execute(self, x):
+        """Expansion of the data.
+
+        :param x: The data to be expanded. Observations/samples must
+            be along the first axis, different variables along the second.
+        :type x: numpy.ndarray
+
+        :returns: The expansion of x with observations/samples along the
+            first axis and corresponding function values (expansion)
+            along the second axis.
+        :rtype: numpy.ndarray
+        """
+
+        num_vars = x.shape[1]
+        num_samples = x.shape[0]
+
+        _with0 = hasattr(self, "with0") and self.with0
+        deg = self._degree
+        dim = self.expanded_dim(num_vars)
+        dim += 1 if not self.with0 else 0
+        result = np.empty(
+            [num_samples, dim], dtype=self.dtype)
+
+        if self.check:
+            self.check_domain(x)
+
+        result[:, 0] = 1.
+        pos = 1
+
+        if self._degree > 1:
+            for cur_var in range(num_vars):
+                # preset index for current variable
+                pos, n, special = self.r_init(result, x, pos, cur_var)
+                # single variable recursion
+                while n <= deg:
+                            # recursion step
+                    result[:, pos] = self.recf(
+                        result, x, special, n, cur_var, pos)
+                    n += 1
+                    pos += 1
+        # in case input is unreasonable
+        else:
+            result[:, 0] = 1
+            for i in range(num_vars):
+                result[:, i+1] = x[:, i]
+
+        todoList = []
+        for i in range(1, num_vars):
+            todoList.append((i*deg, i, 1))
+        # compute the rest of the "simplex" product
+        while len(todoList) > 0:
+            # pos = process(*todoList.pop(0), deg, pos, result, todoList)
+            pos = process(*todoList.pop(0)+(deg, pos, result, todoList))
+        return (result if _with0 else result[:, 1:])
+
+    def check_domain(self, x, prec=1e-6):
+        """Checks for compliance of the data x with the domain on which
+            the function sequence selected is defined or orthogonal.
+
+        :param x: The data to be expanded. Observations/samples must
+            be along the first axis, different variables along the second.
+        :type x: numpy.ndarray
+
+        :param prec: (Numerical) tolerance when checking validity.
+        :type prec: float
+
+        :raise mdp.NodeException: If one or more values lie outside of the function
+            specific domain.
+        """
+        xmax = np.amax(x)-prec
+        xmin = np.amin(x)+prec
+
+        if (self.upper < xmax) or (self.lower > xmin):
+            raise mdp.NodeException(
+                "One or more values lie outside of the function specific domain.")
+
+    @staticmethod
+    def is_trainable():
```

Comment:
Is this function needed? Insn't False the default already?

Reply:
It is. I omitted the methods in the new version.